httplib.h

```diff
@@ -8361,7 +8363,7 @@ inline SSLClient::SSLClient(const std::string &host, int port,
     : ClientImpl(host, port, client_cert_path, client_key_path) {
   ctx_ = SSL_CTX_new(TLS_client_method());
 
-  detail::split(&host_[0], &host_[host_.size()], '.',
+  detail::split(host_.data(), &host_[host_.size()], '.',
```

Comment:
If we do `host.data()`, should we do the same to `&host[host.size()]` like `host_data() + host.size()` This change actually creates inconsistency and made the code less readable...

Reply:
Good point, I overlooked this case.

Reply:
I am ok with the notation like `&a[0]`. IMHO, most of C/C++ programmers know what it means, and it's a common notation.

Reply:
I agree with you, the meaning is very well-known. In this particular case, it even makes more sense to use this notation because of the begin/end parameters of `detail::split()` for iteration. In the other cases, the meaning really is to put data into a vector's buffer.